index.js

```diff
@@ -5,52 +5,62 @@ var _s = require("underscore.string");
 
 // Based on WebInspector.CPUProfileView in CPUProfileView.js in Blink source.
 // https://github.com/yoavweiss/Blink/blob/master/Source/devtools/front_end/CPUProfileView.js
-var calculateTimes = function(profile) {
-    var totalHitCount = function(node) {
-        var result = node.hitCount;
-        for (var i = 0; i < node.children.length; i++) {
-            result += totalHitCount(node.children[i]);
-        }
-        return result;
-    };
+var totalHitCount = function (node) {
+    var result = node.hitCount;
+    for (var i = 0; i < node.children.length; i++) {
+        result += totalHitCount(node.children[i]);
+    }
+    return result;
+};
+
+var calculateTimesForNode = function (node, samplingInterval) {
+    node.selfTime = node.hitCount * samplingInterval;
+    node.selfHitCount = node.hitCount;
+    var totalHitCount = node.hitCount;
+    for (var i = 0; i < node.children.length; i++) {
+        totalHitCount += calculateTimesForNode(node.children[i], samplingInterval);
+    }
+    node.totalTime = totalHitCount * samplingInterval;
+    node.totalHitCount = totalHitCount;
+    return totalHitCount;
+};
+
+var calculateTimes = function (profile) {    
     profile.totalHitCount = totalHitCount(profile.head);
     profile.totalTime = 1000 * (profile.endTime - profile.startTime);
-
     var samplingInterval = profile.totalTime / profile.totalHitCount;
 
-    var calculateTimesForNode = function(node) {
-        node.selfTime = node.hitCount * samplingInterval;
-        node.selfHitCount = node.hitCount;
-        var totalHitCount = node.hitCount;
-        for (var i = 0; i < node.children.length; i++) {
-            totalHitCount += calculateTimesForNode(node.children[i]);
-        }
-        node.totalTime = totalHitCount * samplingInterval;
-        node.totalHitCount = totalHitCount;
-        return totalHitCount;
-    };
-    calculateTimesForNode(profile.head);
+    calculateTimesForNode(profile.head, samplingInterval);
 };
 
-var walkTree = function(node, cb) {
-    if (!node) {
-        return;
+var treeToArrayAcc = function (node, acc) {
+    acc.push(node);
+    for (var i = 0; i < node.children.length; i++) {
+        acc = acc.concat(treeToArrayAcc(node.children[i], []));
     }
-    cb(node);
-    if (!node.children) {
-        return;
-    }
-    node.children.forEach(function(child) {
-        walkTree(child, cb);
-    });
+    return acc;
+}
+
+var treeToArray = function (node) {
+    return treeToArrayAcc(node, []);
+}
+
+var fnForCall = function(call) {
```

Comment:
I wasn't writing this originally for high performance on massive profiles, but imagine that this bit could be optimized a lot more (particularly the `_s.sprintf`)